packages/signers/src/turnkey-secp256k1-signer.ts

```diff
@@ -0,0 +1,124 @@
+
+if (!ORG_ID || !API_PUBLIC_KEY || !API_PRIVATE_KEY || !ECDSA_KEY_ID) {
+    throw new Error("Turnkey environment variables are not set. Please create a .env file.");
+}
+
+// Helper function to wait for a certain amount of time.
+const wait = (ms: number) => new Promise(r => setTimeout(r, ms));
+
+export class TurnkeySecp256k1Signer implements Signer {
+    private tk: TurnkeyClient;
+    public readonly _publicKey: Uint8Array;
+
+    private constructor(
+        publicKey: Uint8Array,
+        turnkeyClient: TurnkeyClient
+    ) {
+        this._publicKey = publicKey;
+        this.tk = turnkeyClient;
+
+        // Bind `this` to the methods to ensure the context is not lost
+        // when they are called by the Sovereign SDK.
+        this.sign = this.sign.bind(this);
+    }
+
+    public static async create(): Promise<TurnkeySecp256k1Signer> {
+        if (!/^[0-9a-fA-F]+$/.test(API_PRIVATE_KEY!)) {
+            throw new Error(`Invalid TURNKEY_API_PRIV: not a valid hex string. Please check your .env file.`);
+        }
+        if (!/^[0-9a-fA-F]+$/.test(API_PUBLIC_KEY!)) {
+            throw new Error(`Invalid TURNKEY_API_PUB: not a valid hex string. Please check your .env file.`);
+        }
+
+        const stamper = new ApiKeyStamper({
+            apiPublicKey: API_PUBLIC_KEY!,
+            apiPrivateKey: API_PRIVATE_KEY!,
+        });
+
+        const turnkeyClient = new TurnkeyClient({ baseUrl: "https://api.turnkey.com" }, stamper);
+
+        const response = await turnkeyClient.getPrivateKey({
+            organizationId: ORG_ID!,
+            privateKeyId: ECDSA_KEY_ID!,
+        });
+
+        const uncompressedPublicKey = response.privateKey.publicKey;
+        if (!uncompressedPublicKey) {
+            throw new Error(`Could not retrieve public key for Turnkey key ${ECDSA_KEY_ID}`);
+        }
+        
+        // The public key from Turnkey is uncompressed (prefixed with 0x04, 65 bytes).
+        // The Sovereign SDK expects a compressed (33-byte) public key.
+        const point   = Point.fromHex(uncompressedPublicKey);
+        const compressedPublicKey = point.toRawBytes(true);
+
+        return new TurnkeySecp256k1Signer(compressedPublicKey, turnkeyClient);
+    }
+
+    public async publicKey(): Promise<Uint8Array> {
+        return this._publicKey;
+    }
+
+    async sign(message: Uint8Array): Promise<Uint8Array> {
+        const msgHash = bytesToHex(keccak_256(message));
+
+        const response = await this.tk.signRawPayload({
+            type: "ACTIVITY_TYPE_SIGN_RAW_PAYLOAD_V2",
+            organizationId: ORG_ID!,
+            timestampMs: String(Date.now()),
+            parameters: {
+                signWith: ECDSA_KEY_ID!,
+                payload: msgHash,
+                encoding: "PAYLOAD_ENCODING_HEXADECIMAL",
+                hashFunction: "HASH_FUNCTION_NO_OP",
+            },
+        });
+
+        // Poll every 60s until COMPLETED
+        while (true) {
```

Comment:
What happens if the program that's running this exits or crashes, etc? We currently have no way to resume a sign request or recover